### DISCUSSION TITLE
Docs improvement: Add page describing how to add attributes

### DISCUSSION THREAD

**Discussion by WhitWaldo** [metalama/Metalama#140] [Ideas]
I might recommend this be placed in a new page under "Advising code" in the conceptual documentation and organized like the "Adding initializers" page (assuming there's a non-programmatic way to do it).

While trying to work my way through doing it myself, I correctly guessed that I could get an `IConstructor` from an `INamedType`, but given that this requires a cast from the `TypeFactory` to an `INamedType`, this might not be the most obvious to someone new to the framework.  For example, there's a lot here that takes some familiarity with Metalama to piece together what it's talking about:

```cs
var attr = AttributeConstruction.Create(
            myAttr.Constructor,
            new List<TypedConstant>
            {
                TypedConstant.Default(typeof(int)),
                TypedConstant.Default(typeof(int))
            },
            new List<KeyValuePair<string, TypedConstant>>());
```

And then one need to know how to actually attach that to the target, needing to know that they use the `AddAttribute()` method in the I*Builder while introducing the type.
'
I might recommend including a link to the aspect ordering page as a reminder that if they attach another aspect, they'll need to factor in ordering somewhere.

This would at least simplify understanding of some of the more advanced Metalama functionality.

**Comment by gfraiteur** [comment]
Done. https://doc.metalama.net/conceptual/aspects/advising/attributes
